Sources/Datadog/FeaturesIntegration/LoggingWithRUMIntegration.swift

```diff
@@ -0,0 +1,42 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+/// Global `RUMMonitor` (if registered).
+private var rumMonitor: RUMMonitor? { RUMMonitor.shared }
+
+/// Provides the current RUM context attributes to produced Logs.
+internal struct LoggingWithRUMContextIntegration {
+    internal struct RUMLogAttributes {
+        static let applicationID = "application_id"
+        static let sessionID = "session_id"
+        static let viewID = "view.id"
+    }
+
+    /// Produces `Log` attributes describing the current RUM context.
+    /// Returns `nil` and prints warning if global `RUMMonitor` is not registered.
+    var currentRUMContextAttributes: [String: Encodable]? {
+        guard let rumContext = rumMonitor?.contextProvider.context else {
+            userLogger.warn("No `RUMMonitor` is registered, so RUM integration with Logging will not work.")
+            return nil
+        }
+
+        return [
+            RUMLogAttributes.applicationID: rumContext.rumApplicationID,
+            RUMLogAttributes.sessionID: rumContext.sessionID.rawValue.uuidString.lowercased(),
+            RUMLogAttributes.viewID: rumContext.activeViewID?.rawValue.uuidString.lowercased(),
+        ]
+    }
+}
+
+/// Creates RUM Errors for Logs.
+internal struct LoggingWithRUMErrorsIntegration {
```

Comment:
do we have a case where we use `LogWithRUMContext` without `LogWithRUMErrors`? can't we merge them into a single type for the sake of simplicity?

Reply:
Yes. The `LogWithRUMContext` can be enabled/disabled with
```swift
logger.builder()
   .bundleWithRUM(...)
   .build()
```
but `LogWithRUMErrors` is always enabled. Also, I think the `LogWithRUMContext` will be needed for `span.log()` integration (when log is created through `span.log()`, the RUM context should also be injected).